src/QMCDrivers/RMC/RMCUpdatePbyP.cpp

```diff
@@ -153,13 +153,11 @@ void RMCUpdatePbyPWithDrift::advanceWalkersVMC()
       dr += sqrttau * deltaR[iat];
       RealType rr = tauovermass * dot(deltaR[iat], deltaR[iat]);
       rr_proposed += rr;
-      if (rr > m_r2max)
+      if (!W.makeMoveAndCheck(iat, dr) || rr > m_r2max)
```

Comment:
No function calls like this in if statements please. The execution order is not obvious. Split.